library/agent/context/isProtectionOffForRoute.ts

```diff
@@ -0,0 +1,24 @@
+import type { Agent } from "../Agent";
+import { updateContext, type Context } from "../Context";
+
+export function isProtectionOffForRoute(
+  agent: Agent,
+  context: Readonly<Context> | undefined
+): boolean {
+  if (!context) {
+    return false;
```

Comment:
don't we want to return true, if context is not set, we don't have userinput to compare to?

Reply:
The existing behavior was to still continue without context at the locations where this function is used.